src/ramanchada2/misc/utils/argmin2d.py

```diff
@@ -145,3 +146,162 @@ def match_peaks(spe_pos_dict, ref):
                 clusters = np.append(clusters, cluster)
     sort_indices = np.argsort(x_spe)
     return (x_spe[sort_indices], x_reference[sort_indices], x_distance[sort_indices], df)
+
+
+def cost_function_position(p1, p2, order_weight=1.0, priority_weight=1.0):
+    order_penalty = order_weight * abs(p1[0] - p2[0])
+    return order_penalty
+
+
+def cost_function(p1, p2, order_weight=1.0, priority_weight=.1):
+    """
+    Modified cost function with an order preservation penalty and priority weighting.
+    - `order_weight` increases penalty for large differences in the x-axis values.
+    - `priority_weight` decreases the cost for higher values in the y-axis for set_b points.
+    """
```

Comment:
It might be nice to follow the agreed upon docstring format: https://docs.python.org/3/library/stdtypes.html#str.isascii